apps/script/checkEntryPoints.js

```diff
@@ -0,0 +1,224 @@
+var chalk = require('chalk');
+var child_process = require('child_process');
+var path = require('path');
+var fs = require('fs');
+
+/**
+ * Configuration for entry points. Mostly this just silences all the bad entry points
+ * we have right now.
+ */
+const SILENCE_OUT_OF_SITES = [
+  'applab-api',
+  'tutorialExplorer',
+  'makerlab',
+  'pd',
+  'publicKeyCryptography',
+  'brambleHost',
+];
+const SITES_CONFIG = {
+  'studio': {
+    templateRoot: '../dashboard/app/views',
+    templateGlobs: ['**/*.erb', '**/*.haml'],
+    templateExtensions: ['erb', 'haml'],
+    silence: [
+      'applab',
+      'applab-api',
+      'bounce',
+      'brambleHost',
+      'calc',
+      'code-studio',
+      'craft',
+      'districtDropdown',
+      'embedBlocks',
+      'embedVideo',
+      'eval',
+      'flappy',
+      'gamelab',
+      'jigsaw',
+      'levelbuilder',
+      'levelbuilder_applab',
+      'levelbuilder_edit_script',
+      'levelbuilder_gamelab',
+      'levelbuilder_markdown',
+      'levelbuilder_studio',
+      'levels/contract_match',
+      'levels/dashboardDialogHelper',
+      'levels/external',
+      'levels/levelGroup',
+      'levels/multi',
+      'levels/textMatch',
+      'levels/widget',
+      'maker/dependencies',
+      'makerlab',
+      'makerlab/setupPage',
+      'maze',
+      'netsim',
+      'pd',
+      'plc',
+      'publicKeyCryptography',
+      'publicKeyCryptography',
+      'raceInterstitial',
+      'schoolInfo',
+      'scriptOverview',
+      'signup',
+      'studio',
+      'termsInterstitial',
+      'turtle',
+      'tutorialExplorer',
+      'weblab',
+    ],
+  },
+  'code.org': {
+    templateRoot: '../pegasus/sites.v3/code.org/public',
+    templateGlobs: ['**/*.erb', '**/*.haml'],
+    templateExtensions: ['erb', 'haml'],
+    silence: [
+      'learn/index',
+    ]
+  },
+};
+
+const ENTRY_POINT_FILE_PATH_PATTERN = /^\.\/src\/sites\/([\w.]+)\/pages\/(.*)\.jsx?$/;
+
+module.exports = function (webpackConfig, options={verbose: false}) {
+  const stats = {
+    failed: 0,
+    silenced: 0,
+    passed: 0,
+  };
+
+  let logVerbose = () => undefined;
+  let logError = console.log;
+  if (options.verbose) {
+    logVerbose = console.log;
+  }
+
+  let entryPromises = Object.keys(webpackConfig.entry).map(key => {
+    let promise = new Promise(resolve => resolve());
```

Comment:
sometimes when I'm writing command line scripts, i switch to vars because I think it will have to run with a really old version of node. But then later I remember that I get to use a newer version of node! That supports let and const! and then I switch but don't go back to fix all the other places.